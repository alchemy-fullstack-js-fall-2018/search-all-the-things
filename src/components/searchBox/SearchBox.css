.searchForm {
  display: grid;
  grid-template-columns: 300px 300px;
  margin: 0px 15%;
}
.searchForm label {
  justify-self: end;
  margin: 5px;
}

.searchForm input {
  justify-self: start;
  margin: 5px;
}
.searchForm input[type=text] {
  width: 100%;
}

.searchForm select {
  justify-self: start;
  margin: 5px;
}
