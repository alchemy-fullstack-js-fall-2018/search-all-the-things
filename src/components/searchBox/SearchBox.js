import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

import styles from './SearchBox.css';

export default class App extends Component {
  static propTypes = {
    query: PropTypes.string.isRequired,
    pageSize: PropTypes.number.isRequired,
    onChange: PropTypes.func.isRequired,
    onPageSize: PropTypes.func.isRequired,
    onSearch: PropTypes.func.isRequired
  };


  render() {


    const { query, pageSize, onChange, onPageSize, onSearch } = this.props;


    const pageSizeOptions = [5, 10, 20, 50]
      .map(size => <option key={size} value={size}>{size}</option>);


    return (
      <Fragment>
        <h2>Search the news!</h2>
        <form>
          {query && <p>search: {query}</p>}
          <div className={styles.searchForm}>
            <label htmlFor="query">search term</label>
            <input
              name="query" value={query}
              type="text" onChange={onChange}
            ></input>
            <label htmlFor="pageSize">Page size</label>
            <select
              name="pageSize" defaultValue={pageSize}
              type="number" onChange={onPageSize}
            >{pageSizeOptions}
            </select>
          </div>
          <button
            type="submit"
            onClick={onSearch}
          >FIND IT</button>
        </form>

      </Fragment>
    );
  }
}
