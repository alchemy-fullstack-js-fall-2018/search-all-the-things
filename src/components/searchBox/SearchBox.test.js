import SearchBox from './SearchBox';

describe('SearchBox Component', () => {
  it('should be a searchBox', () => {

    const searchBox = new SearchBox({
      query: 'dog',
      pageSize: 20,
      onChange: () => { return null; },
      onPageSize: () => { return null; },
      onSearch: () => { return null; }
    });
    expect(searchBox.render()).toMatchSnapshot();
  });
});
