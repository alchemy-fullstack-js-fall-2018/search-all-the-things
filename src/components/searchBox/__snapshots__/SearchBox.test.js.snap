// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Component should be a searchBox 1`] = `
<React.Fragment>
  <h2>
    Search the news!
  </h2>
  <form>
    <p>
      search: 
      dog
    </p>
    <div
      className="searchForm"
    >
      <label
        htmlFor="query"
      >
        Search terms
      </label>
      <input
        name="query"
        onChange={[Function]}
        type="text"
        value="dog"
      />
      <label
        htmlFor="pageSize"
      >
        Page size
      </label>
      <select
        defaultValue={20}
        name="pageSize"
        onChange={[Function]}
      >
        <option
          value={5}
        >
          5
        </option>
        <option
          value={10}
        >
          10
        </option>
        <option
          value={20}
        >
          20
        </option>
        <option
          value={50}
        >
          50
        </option>
      </select>
    </div>
    <button
      onClick={[Function]}
      type="submit"
    >
      FIND IT
    </button>
  </form>
</React.Fragment>
`;
